### PR TITLE
Implement DW_OP_implicit_pointer and DW_OP_entry_value

### PR DESCRIPTION
**Pull request by philipc**
- move op.rs tests into submodule
- convert operation parsing tests to use `test_assembler`
- implement DW_OP_implicit_pointer and DW_OP_entry_value

The implementation for the new operations is minimal: it offloads all the real work to the caller. We probably want to add more support in future to make it easier for users to handle these though.

I didn't convert the operation evaluation tests to use `test_assembler` because @tromey had already written some code that does basically the same thing, so there wouldn't be much benefit to doing so (other than consistency).

### CODE REVIEW

File: src/op.rs [hunk 1]
```diff
@@ -221,6 +230,13 @@ pub enum Location<'input> {
         /// The value.
         value: &'input [u8],
     },
+    /// The piece is a pointer to a value which has no action location.
```
Comment (tromey) [inline src/op.rs]: I think you probably meant s/action/actual/

File: src/op.rs [hunk 2]
```diff
@@ -771,6 +787,16 @@ impl<'input, Endian> Operation<'input, Endian>
                 Ok((newbytes, Operation::ImplicitValue { data: data.into() }))
             }
             constants::DW_OP_stack_value => Ok((bytes, Operation::StackValue)),
+            constants::DW_OP_implicit_pointer |
+            constants::DW_OP_GNU_implicit_pointer => {
+                let (newbytes, value) = try!(parse_offset(bytes, format));
+                let (newbytes, byte_offset) = try!(parse_signed_lebe(newbytes));
+                Ok((newbytes,
+                    Operation::ImplicitPointer {
+                    value: DebugInfoOffset(value),
```
Comment (tromey) [inline src/op.rs]: The indentation looks off here but I can't tell if that is github or just needing rustfmt.

File: src/op.rs [hunk 3]
```diff
@@ -2485,13 +2518,30 @@ mod tests {
 
         check_eval_with_context(&program, Ok(&result), 8, Format::Dwarf64, context);
 
+        // Test `evaluate_entry_value` callback.
+        let program = [
+            Op(DW_OP_entry_value), Sleb(8), U64(0x12345678),
```
Comment (tromey) [inline src/op.rs]: Pedantically I think this should be Uleb(8)

File: src/op.rs (hunk 1 above)
Comment (philipc) [inline src/op.rs]: Yep, thanks.

File: src/op.rs (hunk 2 above)
Comment (philipc) [inline src/op.rs]: Neither :) Looks like a rustfmt bug, where it's trying to block indent the struct members, but forgets that the struct is also visually indented for the enclosing tuple.

File: src/op.rs (hunk 3 above)
Comment (philipc) [inline src/op.rs]: Fixed, thanks.

**Comment by philipc** [comment]
> This one is too big to see in github, but if it's just a pure move of the tests, then I think that's a good idea.

Yeah it is just a move.

**Pull request merged**